core/src/main/java/org/infinispan/configuration/cache/SitesConfiguration.java

```diff
@@ -165,4 +190,40 @@ public boolean equals(Object obj) {
    public String toString() {
       return "SitesConfiguration [backupFor=" + backupFor + ", allBackups=" + allBackups + ", attributes=" + attributes + "]";
    }
+
+   @SuppressWarnings("rawtypes")
+   private static class MergePolicySerializer extends
+         AttributeSerializer<XSiteEntryMergePolicy, SitesConfiguration, SitesConfigurationBuilder> {
+      @Override
+      public Object getSerializationValue(Attribute<XSiteEntryMergePolicy> attribute,
+            SitesConfiguration configurationElement) {
+         XSiteEntryMergePolicy<?, ?> entryMergePolicy = attribute.get();
+         XSiteMergePolicy mergePolicy = XSiteMergePolicy.fromInstance(entryMergePolicy);
+         return mergePolicy == null ? entryMergePolicy.getClass().getName() : mergePolicy.toString();
+      }
+
+      @Override
+      public Object readAttributeValue(String enclosingElement, AttributeDefinition attributeDefinition,
+            Object attrValue, SitesConfigurationBuilder builderInfo) {
+         assert attrValue instanceof String;
+         return XSiteMergePolicy.instanceFromString((String) attrValue, builderInfo.getClass().getClassLoader());
+      }
+   }
+
+   @SuppressWarnings("rawtypes")
+   private static class MergePolicyAttributeCopier implements AttributeCopier<XSiteEntryMergePolicy> {
+
+      @Override
+      public XSiteEntryMergePolicy copyAttribute(XSiteEntryMergePolicy attribute) {
+         if (attribute == null) {
+            return null;
+         }
+         if (attribute instanceof XSiteMergePolicy) {
+            //the default implementations are immutable and can be reused.
+            return ((XSiteMergePolicy) attribute).getInstance();
```

Comment:
Looks like just `return attribute` would work.

Reply:
I want to use `XSiteMergePolicy` as an alias. When the configuration is created, the `copyAttribute()` is invoked and I want the real implementation.